File: Rentacar.Mobile/Rentacar.Mobile/Rentacar.Mobile/Models/Request/BookVehiclesRequest.cs
using System;
using System.Collections.Generic;

namespace Rentacar.Dto.Request
{
    public class BookVehiclesRequest
    {
        public BookVehiclesRequest()
        {
            VehicleTypes = new List<string>();
        }

        public int? TransmissionType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int? MinPrice { get; set; }
        public int? MaxPrice { get; set; }
        public List<string> VehicleTypes { get; set; }

    }
}

File: rentacar.flutter/lib/models/requests/book_vehicles_request.dart
class BookVehiclesRequest {
  int? transmissionType;
  DateTime? startDate;
  DateTime? endDate;
  int? minPrice;
  int? maxPrice;
  List<String>? vehicleTypes;

  BookVehiclesRequest(this.transmissionType, this.startDate, this.endDate,
      this.maxPrice, this.minPrice, this.vehicleTypes);

  BookVehiclesRequest.fromJson(Map<String, dynamic> bookVehiclesMap) {
    transmissionType = bookVehiclesMap['transmissionType'];
    startDate = DateTime.parse(bookVehiclesMap['startDate']);
    endDate = DateTime.parse(bookVehiclesMap['endDate']);
    minPrice = bookVehiclesMap['minPrice'];
    maxPrice = bookVehiclesMap['maxPrice'];
    vehicleTypes = bookVehiclesMap['vehicleTypes'];
  }

  Map<String, dynamic> toJson() {
    return {
      'transmissionType': transmissionType,
      'startDate': startDate?.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'minPrice': minPrice,
      'maxPrice': maxPrice,
      'vehicleTypes': vehicleTypes,
    };
  }
}
